File: Ahorn/entities/catEye.jl
module CH_CatEye

using ..Ahorn, Maple
@mapdef Entity "CherryHelper/CatEye" CatEye(x::Integer, y::Integer,state::String="Normal",Wiggles::Bool=true,respawnTime::Number=4.0)
states=String[
    "Normal",
    "Double_Dash",
    "Shadow_Dash"   
]
const placements = Ahorn.PlacementDict(
    "Cat Eye ($(replace(state,"_"=>" ")), Cherry Helper)" => Ahorn.EntityPlacement(
        CatEye,
        "rectangle",
        Dict{String, Any}(
            "state" => state
        )
    ) for state in states

)

function Ahorn.selection(entity::CatEye)
    x, y = Ahorn.position(entity)
    sprite = "danger/cateye/cateye00.png"

    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::CatEye, room::Maple.Room)
    if get(entity.data, "state", "Normal") == "Normal"
        theSprite = "danger/cateye/cateye00.png"
    elseif get(entity.data, "state", "Normal") == "Double_Dash"
        theSprite = "danger/twodashcateye/cateye00.png"
    elseif get(entity.data, "state", "Normal") == "Shadow_Dash"
        theSprite = "danger/shadowcateye/cateye00.png"
    end
    sprite =  theSprite
    Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end

File: Loenn/entities/catEye.lua
local catEye = {}

local catEyeOptions = {
    Normal = "Normal",
    Double_Dash = "Double_Dash",
    Shadow_Dash = "Shadow_Dash"
}

catEye.name = "CherryHelper/CatEye"
catEye.justification = {0.5, 0.5}
catEye.fieldInformation = {
    state = {
        options = catEyeOptions,
        editable = false
    }
}
catEye.placements = {}

for _, type in pairs(catEyeOptions) do
    table.insert(catEye.placements, {
        name = type,
        data = {
            state = type,
            Wiggles = true,
            respawnTime = 4.0
        }
    })
end

function catEye.texture(room, entity)
    if entity.state == catEyeOptions.Normal then
        return "danger/cateye/cateye00"
    elseif entity.state == catEyeOptions.Double_Dash then
        return "danger/twodashcateye/cateye00"
    elseif entity.state == catEyeOptions.Shadow_Dash then
        return "danger/shadowcateye/cateye00"
    end
end

return catEye